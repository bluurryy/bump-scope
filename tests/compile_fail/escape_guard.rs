use bump_scope::Bump;

#[allow(unused_assignments)]
fn escape_closure(bump: &mut Bump) {
    let mut escapee = None;

    {
        let mut guard = bump.scope_guard();
        let scope = guard.scope();

        escapee = Some(scope.alloc("escape?"));
    }

    dbg!(escapee);
}

fn main() {}
