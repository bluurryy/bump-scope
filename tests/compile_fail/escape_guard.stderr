error[E0597]: `guard` does not live long enough
  --> tests/compile_fail/escape_guard.rs:10:21
   |
9  |         let mut guard = bump.scope_guard();
   |             --------- binding `guard` declared here
10 |         let scope = guard.scope();
   |                     ^^^^^ borrowed value does not live long enough
...
13 |     }
   |     - `guard` dropped here while still borrowed
14 |
15 |     dbg!(escapee);
   |          ------- borrow later used here
